Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+
+        self.setPropertySettings('QuartzInputWorkspace',
+                                 EnabledWhenProperty(vanadium, sample, LogicOperator.Or))
+
+        self.declareProperty(WorkspaceGroupProperty('VanadiumInputWorkspace', '',
+                                                    direction=Direction.Input,
+                                                    optional=PropertyMode.Optional),
+                             doc='The name of the vanadium workspace.')
+
+        self.setPropertySettings('VanadiumInputWorkspace', EnabledWhenProperty('DetectorEfficiencyCalibration',
+                                                                               PropertyCriterion.IsEqualTo, 'Vanadium'))
+
+        self.declareProperty('AverageScan', False,
+                             doc='Whether or not to average the detector counts from multiple scan steps.')
+        self.setPropertySettings('AverageScan', scan)
+
+        self.declareProperty('SumScan', False,
+                             doc='Whether or not to sum the multiple scan steps into a single distribution in TwoTheta angle.')
+
+        self.setPropertySettings('SumScan', scan)
+
+        self.declareProperty('AbsoluteUnitsNormalisation', True,
+                             doc='Whether or not express the output in absolute units.')
+
+        self.setPropertySettings('AbsoluteUnitsNormalisation', EnabledWhenProperty(vanadium, sample, LogicOperator.Or))
+
+        self.declareProperty('ClearCache', True,
+                             doc='Whether or not to clear the cache of intermediate workspaces.')
+
+        self.declareProperty(name="MeasurementTechnique",
+                             defaultValue="Powder",
+                             validator=StringListValidator(["Powder", "SingleCrystal", "TOF"]),
+                             direction=Direction.Input,
+                             doc="Which measurement technique was used to measure the sample.")
+
+        self.declareProperty(name="ComponentSeparationMethod",
+                             defaultValue="None",
+                             validator=StringListValidator(["None", "Uniaxial", "XYZ", "10p"]),
+                             direction=Direction.Input,
+                             doc="Whether to perform component separation and what type of method to use.")
+
+        self.declareProperty(name="SampleGeometry",
+                             defaultValue="None",
+                             validator=StringListValidator(["None", "FlatPlate", "Cylinder", "Annulus", "Custom"]),
+                             direction=Direction.Input,
+                             doc="Sample geometry for self-attenuation correction to be applied.")
+
+        self.setPropertySettings('SampleGeometry', EnabledWhenProperty(vanadium, sample, LogicOperator.Or))
+
+        self.declareProperty(PropertyManagerProperty('ExperimentPropertiesDictionary', dict()),
```

Comment:
This property is poorly named. Maybe you can rename to just `SampleProperties`?

Reply:
I agree it is rather unfortunately named, however this dictionary can contain information about container (mass, thickness...), beam, etc. needed by PaalmanPings algorithm. It can be changed to `SampleProperties` if you think this is the best option.

Reply:
it's true that it contains sample, environment and beam properties, maybe `SampleAndEnvironmentProperties` would be more bespoke.